File: Server/Model/HouseHold.cs
using System.ComponentModel.DataAnnotations.Schema;
using DTO;

namespace Model;

public class Household
{
    public int Id { get; set; }
    public string Name { get; set; } = "";
    public int Code { get; set; }
    public int UserId { get; set; }

    [NotMapped]
    public List<int> AvailableAvatars { get; set; } = new List<int>();
    
    public ICollection<Profile> Profiles { get; }  = new List<Profile>();
    public ICollection<Chore> Chores { get; }  = new List<Chore>();

    public Household(HouseholdDto dto)
    {    
        Name = dto.Name;
        UserId = dto.OwnerId;
        Random random = new Random();
        Code = random.Next(1000, 9999);        
    }

    public Household()
    {
        
    }
}

File: Client/src/Data/Household.ts
import { User } from "./User";

export interface Household {
  id: number;
  name: string;
  code: number;
  owner: User;
  availableAvatars: number[];
}

export interface AddHouseholdDTO {
  Name: string;
  OwnerId: number;
}

export type JoinHouseholdDto = {
  userId: number;
  householdId: number;
  code: number;
  displayName: string;
  avatar: number;
  isAdmin: boolean;
};
